src/Runner.Worker/StepsRunner.cs

```diff
@@ -195,6 +195,20 @@ public async Task RunAsync(IExecutionContext jobContext)
                         }
                         else
                         {
+                            // This is our last, best chance to expand the display name.  (At this point, all the requirements for successful expansion should be met.)
+                            try
+                            {
+                                if (step is IActionRunner actionRunner && actionRunner.Stage == ActionRunStage.Main &&
+                                    actionRunner.TryEvaluateDisplayName(step.ExecutionContext.ExpressionValues, step.ExecutionContext))
+                                {
+                                    step.ExecutionContext.UpdateTimelineRecordDisplayName(actionRunner.DisplayName);
+                                }
+                            }
+                            catch (Exception ex)
+                            {
+                                Trace.Info("Caught exception while attempting to evaulate/update the step's DisplayName.  Exception Details:  {0}", ex);
```

Comment:
if I got this C# code right, this `Exception` is the 'parent' of SystemException and C#'s docs say `Because SystemException serves as the base class of a variety of exception types, your code should not throw a SystemException exception, nor should it attempt to handle a SystemException exception unless you intend to re-throw the original exception.`. Would at least using `Trace.Error` in this case be better or not catching this generic exception at all? 🤔

Reply:
> or not catching this generic exception at all

+1 for not using a `try..catch` block here at all, **unless** exceptions during the display name evaluation must explicitly be ignored by design/acceptance-criteria.

If there is no concrete exception type thrown by the wrapped code **and** any possible exceptions should not be ignored, it's usually best letting the exception bubbling up to a higher / the topmost handler (which usually does some kind of logging).

Reply:
I considered this.

Evaluation of the condition is paramount.  We have to ensure that that happens no-matter-what.  In other words, we can't allow a failure to update the display name (which would be benign in the grand scheme of things) to derail evaluation of the condition.

Reply:
@flobernd with the correct *unless* - from the issue / reqs doc:

> we should ensure that this evaluation cannot fail (skipped steps should never turn to failed and cause a job to fail that would have otherwise succeeded).

We really _do_ want to make this a best / last effort display name calculation.
Let's make sure however that we don't use `catch (Exception ex)` for flow handling, like catching ArgUtil.NotNull exceptions. For example, could `actionRunner.DisplayName` be `null` at the time of calling `step.ExecutionContext.UpdateTimelineRecordDisplayName`?